transport/http2_client.go

```diff
@@ -601,7 +608,7 @@ func (t *http2Client) Close() (err error) {
 	}
```

Comment:
This `if` will always be true.
Actually, is `errorChan === t.ctx.Done()`?

Reply:
Removed the if, and unreachable entirely.

Yes, errorChan is essentially t.ctx.Done(), with only subtly different semantics that were irrelevant.  Removed.  Good catch.